server/src/test/java/org/opensearch/extensions/rest/RestExecuteOnExtensionTests.java

```diff
@@ -25,19 +28,38 @@ public class RestExecuteOnExtensionTests extends OpenSearchTestCase {
     public void testRestExecuteOnExtensionRequest() throws Exception {
         Method expectedMethod = Method.GET;
         String expectedUri = "/test/uri";
-        RestExecuteOnExtensionRequest request = new RestExecuteOnExtensionRequest(expectedMethod, expectedUri);
+        PrincipalIdentifierToken expectedRequestIssuerIdentity = new PrincipalIdentifierToken("requestor-identifier");
+        NamedWriteableRegistry registry = new NamedWriteableRegistry(
+            org.opensearch.common.collect.List.of(
+                new NamedWriteableRegistry.Entry(
+                    PrincipalIdentifierToken.class,
+                    PrincipalIdentifierToken.NAME,
+                    PrincipalIdentifierToken::new
```

Comment:
`PrincipalIdentifierToken` doesn't have a no-arg constructor, does this code work?

Reply:
The test was executed locally and passed.  To registry an entry it needs a `(InputStream) -> T` function, see [Writeable](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/io/stream/Writeable.java#L94) - we got the opportunity to learn all about NamedWriteables when authoring this code :D